hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CustomizedScanInfoBuilder.java

```diff
@@ -80,4 +93,19 @@ public KeepDeletedCells getKeepDeletedCells() {
     return keepDeletedCells != null ? keepDeletedCells : scanInfo.getKeepDeletedCells();
   }
 
+  @Override
+  public int getMinVersions() {
+    return minVersions != null ? minVersions : scanInfo.getMinVersions();
+  }
+
+  @Override
+  public void setMinVersions(int minVersions) {
+    this.minVersions = minVersions;
+  }
+
+  @Override
+  public Scan getScan() {
+    return scan;
```

Comment:
Changes made on the returned scan object will be effective and visible beyond the caller. Clone and return?

Reply:
I didn't check to see how much of Scan is immutable...

Reply:
Scan is mostly mutable, I think. That's a good point -- the clone in the constructor will prevent changes from executing, but a caller could still mess up the state for future callers. I'll clone in the getter.

Reply:
Done.